sbin/pfdns

```diff
@@ -214,8 +220,78 @@ sub response_handler {
            return  regzone($qname, $qclass, $qtype, $peerhost,$query,$conn);
         }
     }
+    
+    foreach my $network (@routed_dns_enforcement_nets_named) {
+        if ($network->contains($ip)) {
+            $logger->trace("Performing DNS enforcement on $ip");
+            my $mac = pf::iplog::ip2mac($ip_addr);
+
+            unless($mac){
+                $logger->warn("Can't lookup IP $ip_addr to a MAC address. Returning result as if unreg.");
+                return regzone($qname, $qclass, $qtype, $peerhost,$query,$conn);
+            }
+
+            my $node = node_attributes($mac);
+            if($node->{status} eq $STATUS_REGISTERED){
+                $logger->info("Node $mac is registered, returning resolved domain.");
+                return resolve($qname, $qclass, $qtype, $peerhost, $query, $conn);
+            }
+            else {
+                $logger->info("Node $mac is unregistered, returning portal domain.");
+                return regzone($qname, $qclass, $qtype, $peerhost,$query,$conn);
+            }
+        }
+    }
+
+    $logger->debug("$ip_addr did not match any known networks. Not responding to DNS request.");
 }
 
+=head2 resolve
+
+Resolve a domain using the operating system configured DNS servers and return it as an answer
+
+=cut
+
+sub resolve {
```

Comment:
Not really used to this piece of code but from what I understand, simply removing `next unless $rr->type eq "A";` below would do it, right ?